apps/web/frontend/components/workHourComment.tsx

```diff
@@ -0,0 +1,119 @@
+import { format, max, min, parseISO } from 'date-fns'
+import { useRef } from 'react'
+import { FaRegCommentDots } from 'react-icons/fa6'
+import { useMutation } from 'urql'
+
+import { toastError } from '@progwise/timebook-ui'
+
+import { FragmentType, graphql, useFragment } from '../generated/gql'
+
+export const WorkHourCommentFragment = graphql(`
+  fragment WorkHourCommentFragment on Task {
+    id
+    title
+    workHourOfDays(from: $from, to: $to) {
+      date
+      workHour {
+        id
+        comment
+      }
+      isLocked
+    }
+  }
+`)
+
+interface WorkHourCommentProps {
+  comment: FragmentType<typeof WorkHourCommentFragment>
+}
+
+const CommentCreateMutationDocument = graphql(`
+  mutation commentUpdate($comment: String!, $date: Date!, $taskId: ID!) {
+    workHourCommentUpdate(date: $date, taskId: $taskId, comment: $comment) {
+      comment
+    }
+  }
+`)
+
+export const WorkHourComment = ({ comment: commentFragment }: WorkHourCommentProps) => {
+  const dialogReference = useRef<HTMLDialogElement>(null)
+  const openDialog = () => {
+    dialogReference.current?.showModal()
+  }
+
+  const task = useFragment(WorkHourCommentFragment, commentFragment)
+  const [, commentUpdate] = useMutation(CommentCreateMutationDocument)
+  const handleBlur = (date: string) => async (event: React.FocusEvent<HTMLTextAreaElement>) => {
+    try {
+      const result = await commentUpdate({
+        comment: event.target.value,
+        date: date,
+        taskId: task.id,
+      })
+      if (result.error) {
+        throw new Error(`GraphQL Error ${result.error}`)
+      }
+    } catch {
+      toastError('Error, comment was not saved')
+    }
+  }
+
+  const firstDay = min(task.workHourOfDays.map((workHourOfDay) => parseISO(workHourOfDay.date)))
+  const lastDay = max(task.workHourOfDays.map((workHourOfDay) => parseISO(workHourOfDay.date)))
+  const dateTimeFormat = new Intl.DateTimeFormat('en-US', { month: 'long', day: 'numeric' })
+
+  const commentCount = task.workHourOfDays.filter((workHourOfDay) => workHourOfDay.workHour?.comment).length
+
+  return (
+    <>
+      <div>
+        <div className="indicator">
+          {commentCount > 0 && (
+            <span className="badge indicator-item badge-secondary badge-xs" title="indicator">
```

Comment:
title should be for 1 comment = "1 comment"
title should be for more than 1 comment: "x comments"